api/hypershift/v1alpha1/hostedcluster_types.go

```diff
@@ -1574,6 +1574,22 @@ type AzurePlatformSpec struct {
 	SubnetID          string `json:"subnetID"`
 	SubscriptionID    string `json:"subscriptionID"`
 	SecurityGroupID   string `json:"securityGroupID"`
+
+	// MSIClientIDs contains the client IDs related to the managed identities needed for the following control plane
+	// components: cluster-image-registry, cluster-ingress, cluster-storage, and cluster-network operators.
+	//
+	// +optional
+	MSIClientIDs *ControlPlaneManagedServiceIdentities `json:"msiClientIDs,omitempty"`
+}
+
+type ControlPlaneManagedServiceIdentities struct {
+	// AzureCloudProviderMSIClientID is the client ID of a pre-existing managed identity ID of that will be associated
+	// with the azure cloud provider, aka ccm. The managed identity is expected to be in
+	// AzurePlatformSpec.ResourceGroupName.
+	//
+	// +kubebuilder:validation:Required
+	// +required
+	AzureCloudProviderMSIClientID string `json:"azureCloudProviderMSIClientID"`
```

Comment:
Is `MSI` an appropriate term/wording? cc Ben Vessel

The reason I am raising this is because according to Azure doc:
https://learn.microsoft.com/en-us/entra/identity/managed-identities-azure-resources/overview

there is a note:
`Managed identities for Azure resources is the new name for the service formerly known as Managed Service Identity (MSI).`

or are we referring to nother concept?

Reply:
I commented in a thread with Bryan, but it might make sense to pull this into a higher-level struct -`AzureResourceManagedIdentities` and then remove MSI from everything.  The higher-level API struct will dictate what the thing is, and we'll separate out the data plane managed identities from the control plane.  

Thoughts?

Reply:
These are in a higher level struct now - `ControlPlaneManagedServiceIdentities`. I was planning to make a high level one for data plane, let's call it `DataPlaneManagedIdentities` for now, ones too when we implement WorkloadIdentity.

Reply:
Side note - is there a known regex for these IDs?